### ISSUE TITLE
ALTER TABLE evo_groups CHANGE COLUMN grp_ID grp_ID                           int(10) UNSIGNED NOT NULL auto_increment

### ISSUE THREAD

**Issue by Metztli** [b2evolution/b2evolution#105]
Stack: Linux, Engine X, MariaDB, PHP (LEMP)
           Linux, Apache2, MariaDB, PHP (LAMP)

Three(3) different MariaDB version, 10.4.10 and 10.5.3, databases, fail to upgrade, manually or automatically, from b2evolution 6.11.7 to b2evolution 7.1.5, as expounded in prior [Converting columns to utf8mb4...](https://github.com/b2evolution/b2evolution/issues/103) issue -- which I closed due to the deafening silence and thus implicit un-acknowledgment the b2evolution installer essentially fails.

All three(3) databases had to be migrated [essentially downgraded] to MySQL 8.0.20 in order to complete _**only** half of the b2evolution upgrades_ , i.e., ironic since most GNU/Linux distributions do not offer MySQL anymore from their repositories and instead host MariaDB. 

This last upgrade I used a Linux node ([Linode](https://www.linode.com)) with KVM hosting an essentially vanilla Debian 10 instance on substandard ext4 file system. I retained the ext4 file system only for the sake of an potentially spurious argument that 'b2evolution installer only works on Linux distribution offered filesystems.'
![linode-ext4-fs-5 6 19-2-reiser4 0 2](https://user-images.githubusercontent.com/3677052/86225946-c921f100-bb3f-11ea-95e8-8818bb84d0f2.png)

Midway through the upgrade, when the database upgrade was being performed, the installer failed again similar that described in #103 with the very first failed entry as
`ALTER TABLE evo_groups CHANGE COLUMN grp_ID grp_ID                           int(10) UNSIGNED NOT NULL auto_increment, DROP PRIMARY KEY`

![b2evo-6 11 7-fail-to-7 1 5_](https://user-images.githubusercontent.com/3677052/86226726-eacfa800-bb40-11ea-837a-a3b8eb6d3af9.png)

Subsequently, manually modifying the above entry by removing ~~auto_increment~~, as
`ALTER TABLE evo_groups CHANGE COLUMN grp_ID grp_ID                           int(10) UNSIGNED NOT NULL, DROP PRIMARY KEY`

and thus successively until the 61 manual entries are completed as elaborated in #103 , then it will be possible for the b2evolution 7.1.5 installer to successfully upgrade the database.

In conclusion b2evolution 7.1.5 installer needs to complete the upgrade by itself since to request to be hand held, literally, by the b2evolution fan is overkill. Additionally, as MariaDB is the database offered by GNU/Linux repositories, the fact that an upgrade from b2evolution 5.x.y to 6.x.y, and then again to 7.x.y, **always fails** is a glaring fact that b2evolution installer needs substantial work.

I won't close this issue because other users will experience this as what it is, a serious b2evolution bug. I am only the messenger here... like Julian Assange :wink: 

P.S. If this sort of failure is a feature, and not a bug, designed to generate subscription revenue -- as someone mentioned to me -- 
we would all benefit from a b2evolution road map, i.e., if due to the nature of the relatively high € subscription entry 'PRO' level, the b2evolution features have as main objective the large enterprises while the casualties are suffered by the community.

Best Professional Regards.

**Comment by fplanque** [comment]
Can you summarize this rant to the shortest way to reproduce your error?

Something that looks like:

1. Use Debian 10
2. Install b2evolution v6.X.Y
3. Try to upgrade to v.X.Y -> BUG > screenshot

**Comment by eumel8** [comment]
I think MySQL doesn't allow DROP PRIMARY KEY during [auto-increment is set](https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html) . From my understanding the upgrade process will drop and add the keys, so [I comment out this part](https://github.com/b2evolution/b2evolution/blob/master/inc/_core/model/db/_upgrade.funcs.php#L705) and the upgrade was ok. 

b2evolution: 6.11.3-stable->7.2.2-stable
MySQL: 5.7 -> 8.0.22
Ubuntu: 18.04 -> 20.04

**Comment by eumel8** [comment]
still the issue while upgrading from 7.2.2-stable to 7.2.3-stable

**Comment by farfallosa** [comment]
@eumel8 Why did you upgrade? 7.2.3 is the latest and last version and there is no support for bugs etc.
I decided to keep versions 6.11.5 and 6.11.7 and there is no need for an upgrade. But may be you experienced some benefits by upgrading and I'd like to know. Please share you experience (if not here, so in Support Forum of b2evolution.net. Would be nice.
Cheers and best for 2022

**Comment by Metztli** [comment]
@eumel8 I did not experience the issue when upgrading from 7.2.2 to 7.2.3 ; the underlying databases were/are MariaDB latest release at the time of the upgrade and now.

As a matter of fact, only the very first upgrade to b2evolution major versions, i.e., 4.x.y to 5.x.y to 6.x.y to 7.x.y, usually introduce major issues at the database level.

Now that b2evolution has been orphaned, the b2evolution installer might benefit from someone (hint: you?) taking over and making the major version upgrades smoother.

Best Professional Regards.

**Comment by eumel8** [comment]
@farfallosa I use auto upgrade feature and 7.2.3 is the latest stable version.  You mention version 6 which remembered me on my install history:

```
b2evolution-6.5.0-stable-2015-06-24.zip
b2evolution-6.7.2-stable-2016-05-15.zip
b2evolution-6.8.2-stable-2016-11-17.zip
b2evolution-6.8.5-stable-2017-01-19.zip
b2evolution-6.9.3-stable-2017-08-25.zip
b2evolution-6.10.2-stable-2018-06-20.zip
b2evolution-6.11.3-stable-2019-08-20.zip
b2evolution-6.11.7-stable-2020-05-01.zip
b2evolution-7.2.2-stable-2020-10-06.zip
b2evolution-7.2.3-stable-2021-03-01.zip
```

Indeed, the problem  starts in version 7, but there are no changes in the db upgrade procedure. I thinks this comes with newer MySQL version.  MySQL 5.5 is still recommended by [b2evolution](https://github.com/b2evolution/b2evolution#requirements) which is EOL since 2018.

I researched a while this MySQL error and come to this conclusion:  https://github.com/b2evolution/b2evolution/pull/112

It's not a big hit because this whole  `_upgrade.funcs.php` needs a rework in my point of view, but I'm also not a MySQL expert :-)

**Comment by eumel8** [comment]
Sometimes later hit by the same error while upgrading 7.2.5-stable. Due the fact, [b2evolution is also gone](https://b2evolution.net/news/2022/03/26/2022-update-eol), are there any forks to continue?
